File: Data Structure/Lab 3/Pascal.c
//WAP to print Pascal triangle.

#include<stdio.h>
void main(){
    int n,a[n][n];

    printf("Enter n :");
    scanf("%d",&n);

    for(int i = 0; i < n; i++){
        for(int k = 0 ; k < n - i; k++){
            printf(" ");
        }
        for(int j = 0; j <= i; j++){
            
            if(j == 0 || j == i){
                a[i][j] = 1;
            }
            else{
                a[i][j] = a[i - 1][j] + a[i - 1][j - 1];
            }
            printf("%d ",a[i][j]);
        }
        printf("\n");
    }
}

File: Data Structure(Java)/Lab 2/Pascal.java
// WAP to print Pascal triangle. 

import java.util.Scanner;

public class Pascal {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of rows :");
        int n = sc.nextInt();

        int[][] arr = new int[n][n];
        
        for(int i = 0; i < n; i++){
            for(int k = 0; k < n - 1; k++){
                System.out.print(" ");
            }
            for(int j = 0 ; j <= i ; j++){
                if(j == 0 || j == i){
                    arr[i][j] = 1;
                }
                else{
                    arr[i][j] = arr[i - 1][j] + arr[i - 1][j - 1];
                }
                System.out.print(arr[i][j]);
            }
            System.out.println();
        }
        
        sc.close();
    }
}
